cpp/include/cugraph/utilities/misc_utils.cuh

```diff
@@ -36,40 +36,40 @@ namespace cugraph {
 
 namespace detail {
 
-template <typename vertex_t, typename edge_t>
-std::tuple<std::vector<vertex_t>, std::vector<edge_t>> compute_offset_aligned_edge_chunks(
+template <typename data_t, typename offset_t>
+std::tuple<std::vector<data_t>, std::vector<offset_t>> compute_offset_aligned_edge_chunks(
   raft::handle_t const& handle,
-  edge_t const* offsets,
-  vertex_t num_vertices,
-  edge_t num_edges,
+  offset_t const* offsets,
+  data_t num_offsets,
+  offset_t num_elements,
```

Comment:
Should we use raw pointers or better switch to device_span?

And we need to double check whether the offset array size is num_vertices or num_vertices + 1 (I assume it should be num_vertices + 1). In this case, offsets array size is now num_offsets + 1. I guess this is a bit misleading.

Reply:
What about something like
```
template <typename major_idx_t, typename minor_idx_t>
std::tuple<std::vector<major_idx_t>, std::vector<minor_idx_t>> compute_offset_aligned_compressed_sparse_array_chunks(
  raft::handle_t const& handle,
  raft::device_span<minor_idx_t> offsets,  // num_majors == offsets.size() - 1
  minor_idx_t num_minors,
  size_t approx_minor_chunk_size)
```

Reply:
Pushing an update that makes the input a device_span.  I named it a bit differently than you suggested.  My use of the function seems (at least to me) less like major/minor, so I left things as `data_t` and `offset_t`, but I tweaked the names to get rid of the notion of vertex and edge.  Let me know what you think.

Reply:
So, here we are partitioning/chunking a compressed sparse array, right?

In case of the all-pairs similarity algorithm, we have seeds and their two hop degree offsets (to index an array to store each seed's two-hop neighbors).

`offset_t` sounds right, but my concern with data_t or element_t is that it is not clear whether this is to index the seeds or the two-hop neighbors.

Not sure what should be the right name for seeds or vertices in chunking compressed sparse array.

Maybe just use vertex_t as seeds are still vertices?